src/client_side_request.h

```diff
@@ -76,6 +76,9 @@ class ClientHttpRequest
     /// the request. To set the virgin request, use initRequest().
     void resetRequest(HttpRequest *);
 
+    /// whether the client sent a CONNECT request to us
```

Comment:
The purpose of this method is still a bit vague to me. On the one hand, it does not represent only static information, related to Squid ports (because it may return, e.g., dynamic flags.forceTunnel). On the other hand, it does not cover all dynamic cases, when sending CONNECT response is already impossible (some of these cases are covered in tunnel.cc).

Reply:
Yes, it is difficult to define this method precisely due to other out-of-scope design problems. I suggest:

```suggestion
    /// Whether the client sent CONNECT to us, and we have not responded before TunnelStateData
    /// got control. Once in control, TunnelStateData factors in its own state to answer this question.
    /// TODO: (Re)move TunnelStateData::Connection::dirty flag to ConnStateData to simplify this.
```

Or, if you prefer, move this back into TunnelStateData to simplify despite the need to dereference a ClientHttpRequest pointer all the time there.

Reply:
Since this method seems to belong to ClientHttpRequest - I think it is the right thing that we moved this method there. I added the proposed TODO - it is not just about moving the flag from one place to another, but also fixing the existing contexts (e.g., TunnelStateData::notePeerReadyToShovel()) where clientExpectsConnectResponse() is used now without the 'dirty' flag.

Reply:
> the proposed TODO - it is not just about moving the flag from one place to another, but also fixing the existing contexts (e.g., TunnelStateData::notePeerReadyToShovel()) where clientExpectsConnectResponse() is used now without the 'dirty' flag.

To keep TODO comments short and avoid spelling out possibly-wrong and likely-to-become-outdated adjustments, I think we can (and should) assume that the developer addressing this TODO will treat it holistically rather than literally. This includes (without limitation) any necessary post-move or post-removal code fixes/polishing and beyond. For example, I would not be surprised if ConnStateData (or perhaps even Connection or the `fde` class) already has a suitable data member or should gain a more suitable data member than an direct equivalent of the current TunnelStateData::Connection::dirty flag.